plugin/hotswap-agent-spring-boot-plugin/src/main/java/org/hotswap/agent/plugin/spring/boot/listener/PropertySourceChangeListener.java

```diff
@@ -26,25 +29,52 @@
 import org.hotswap.agent.plugin.spring.reload.BeanChangeEvent;
 import org.hotswap.agent.util.AnnotationHelper;
 import org.hotswap.agent.util.spring.util.ClassUtils;
+import org.hotswap.agent.util.spring.util.ObjectUtils;
 import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
 import org.springframework.beans.factory.support.DefaultListableBeanFactory;
 import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.ConfigurableApplicationContext;
 
 /**
  * PropertySourceChangeBootListener: refresh configuration properties when property source changed.
  * (Not refresh directly but send a BeanChangeEvent to refresh ConfigurationProperties bean)
  */
-public class PropertySourceChangeBootListener implements SpringListener<SpringEvent<?>> {
+public class PropertySourceChangeListener implements SpringListener<SpringEvent<?>> {
+
+    private static AgentLogger LOGGER = AgentLogger.getLogger(PropertySourceChangeListener.class);
+
+    private static Map<ConfigurableListableBeanFactory, PropertySourceChangeListener> instances = new HashMap<>(4);
```

Comment:
I think we should do further to eliminate 'instances'.

Reply:
Please review it again.